linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/CoreJSLib.scala

```diff
@@ -725,6 +725,37 @@ private[emitter] object CoreJSLib {
     private def defineDispatchFunctions(): Tree = {
       val instance = varRef("instance")
 
+      val methodsInRepresentativeClasses = globalKnowledge
+        .methodsInRepresentativeClasses()
+        .filter(_._2 != toStringMethodName) // toString() is special
+        .groupBy(_._2)
+        .map(kv => (kv._1, kv._2.map(_._1)))
+        .toList
+        .sortBy(_._1.nameString) // sort for determinism
+
+      /* Hijacked classes ordered by priority for the dispatch, except "small"
+       * number types (which are covered by Double).
+       * Those with `typeof`-based tests come first.
+       */
+      val nonSmallNumberHijackedClasses = List(
+        BoxedDoubleClass,
+        BoxedStringClass,
+        BoxedBooleanClass,
+        BoxedUnitClass,
+        BoxedCharacterClass
+      ) ::: (
+        if (allowBigIntsForLongs) List(BoxedLongClass) else Nil
+      ) ::: List(
+        BoxedCharacterClass
+      )
```

Comment:
I do not understand why we need this explicit list. IIUC, we already partition these based on whether they have a `typeOf` test anyways.

Reply:
We do partition, but if there is only one with a `typeof` test (or 0), we take the full list again, so it's still better that the ones with `typeof` come first.